src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -132,10 +146,17 @@ private static TypeSymbol InferReturnType(BoundBlock block, Binder binder, bool
 
             // Async:
 
+            var delegateReturnType = delegateType?.GetDelegateType()?.DelegateInvokeMethod?.ReturnType as NamedTypeSymbol;
+            var asyncMethodBuilder = (object)delegateReturnType == null ?
+                AsyncMethodBuilder.NotAsync :
+                binder.GetAsyncMethodBuilder(block.Syntax, delegateReturnType);
+
             if (resultTypes.IsEmpty)
             {
                 // No return statements have expressions; inferred type Task:
```

Comment:
Adjust this comment "inferred type either Task or, if delegate was supplied, ..."